### DISCUSSION TITLE
I cannot use ddu#pop()

### DISCUSSION THREAD

**Discussion by hal-ito** [Shougo/ddu.vim#78] [Q&A] [answered]
Thank you for the nice supports.

Below code is now working.
```
nnoremap <Leader><Leader>g <Cmd>call ddu#start({'name': 'grep', 'push': v:true, 'sources':[{'name': 'rg', 'params': {'input': expand('<cword>')}}], })<CR>
```

But I still stumbling next step. below:
```
nnoremap <silent> <Leader>n <Cmd>call ddu#pop('grep')<CR><Cmd>call ddu#ui#multi_actions([['cursorNext'], ['itemAction']])<CR>
nnoremap <silent> <Leader>p <Cmd>call ddu#pop('grep')<CR><Cmd>call ddu#ui#multi_actions([['cursorPrevious'], ['itemAction']])<CR>
```

I tryied `:call ddu#pop('grep')`.  But no responses and no error messages.

**Comment by Shougo** [comment]
What do you want to do by the actions?

I don't understand your desired.

I think you don't know about ddu features.

`ddu#pop()` is to restore the previous session, but you don't have the session.  So it is ignored.

I think `resume` option is better for it instead of `push`.

`push` means save current session and create new session.
The feature is not for user use.  It is added for UI.

**Reply by Shougo** [reply]
For example, I use this.

```vim
nnoremap ;g <Cmd>Ddu
      \ -name=search rg -resume=v:false
      \ -ui-param-ff-ignoreEmpty
      \ -source-param-rg-input='`'Pattern: '->input('<cword>'->expand())`'
      \ <CR>
nnoremap n <Cmd>Ddu
      \ -name=search -resume
      \ -ui-param-ff-startFilter=v:false
      \ <CR>
```

**Reply by hal-ito** [reply]
Thank you for your kindness supports. and very nice example.

I want to use Ddu instead of :grep and quickfix.
So I want to jump to next / previous match by Leader-n / Leader-p.

It was so useful setting when I used Denite. So I want to use Ddu same looks.

If you don't mind help me?

**Reply by Shougo** [reply, marked as answer]
You can use this.

```vim
	nnoremap <C-n>
	      \ <Cmd>call ddu#ui#multi_actions(
	      \   ['cursorNext', 'itemAction'], 'grep')<CR>
	nnoremap <C-p>
	      \ <Cmd>call ddu#ui#multi_actions(
	      \   ['cursorPrevious', 'itemAction'], 'grep')<CR>
```

NOTE:  plugins must be the latest.

**Reply by hal-ito** [reply]
I tried this setting at last year, but I could not it.
It makes me so happy. Thank you!
